app/src/main/java/com/kickstarter/viewmodels/MessageHolderViewModel.java

```diff
@@ -0,0 +1,119 @@
+package com.kickstarter.viewmodels;
+
+import android.support.annotation.NonNull;
+import android.support.v4.util.Pair;
+
+import com.kickstarter.R;
+import com.kickstarter.libs.ActivityViewModel;
+import com.kickstarter.libs.CurrentUserType;
+import com.kickstarter.libs.Environment;
+import com.kickstarter.models.Message;
+import com.kickstarter.models.User;
+import com.kickstarter.services.ApiClientType;
+import com.kickstarter.ui.viewholders.MessageViewHolder;
+
+import rx.Observable;
+import rx.subjects.PublishSubject;
+
+import static com.kickstarter.libs.rx.transformers.Transformers.combineLatestPair;
+import static com.kickstarter.libs.rx.transformers.Transformers.neverError;
+
+public interface MessageHolderViewModel {
+
+  interface Inputs {
+    /** Call to configure the view model with a message. */
+    void configureWith(Message message);
+  }
+
+  interface Outputs {
+    /** Emits a boolean that determines whether the creator's avatar image should be hidden. */
+    Observable<Boolean> creatorAvatarImageHidden();
+
+    /** Emits the url for the creator's avatar image. */
+    Observable<String> creatorAvatarImageUrl();
+
+    /** Emits a boolean to determine whether the message body view should align the parent view's end and right. */
+    Observable<Boolean> messageBodyTextViewAlignParentEnd();
+
+    /** Emits the color int for the message background. */
+    Observable<Integer> messageBodyTextViewBackgroundColorInt();
+
+    /** Emits the message body text view text. */
+    Observable<String> messageBodyTextViewText();
+
+    /** Emits the color int for the message text. */
+    Observable<Integer> messageBodyTextViewTextColorInt();
+  }
+
+  final class ViewModel extends ActivityViewModel<MessageViewHolder> implements Inputs, Outputs {
+    private final ApiClientType client;
+    private final CurrentUserType currentUser;
+
+    public ViewModel(final @NonNull Environment environment) {
+      super(environment);
+
+      this.client = environment.apiClient();
+      this.currentUser = environment.currentUser();
+
+      final Observable<User> freshUser = this.client.fetchCurrentUser()
+        .retry(2)
+        .compose(neverError());
```

Comment:
heyooo, don't think we should be refreshing the user from a view holder! this would execute multiple times based on how many messages are visible at once.

whatcha tryna do with this jazz?